sqlglot/dialects/bigquery.py

```diff
@@ -196,6 +196,19 @@ def _ts_or_ds_diff_sql(self: BigQuery.Generator, expression: exp.TsOrDsDiff) ->
     return self.func("DATE_DIFF", expression.this, expression.expression, unit)
 
 
+def _unix_to_time_sql(self: BigQuery.Generator, expression: exp.UnixToTime) -> str:
+    scale = expression.args.get("scale")
+    timestamp = self.sql(expression, "this")
+    if scale == exp.UnixToTime.SECONDS:
+        return f"TIMESTAMP_SECONDS({timestamp})"
+    if scale == exp.UnixToTime.MILLIS:
+        return f"TIMESTAMP_MILLIS({timestamp})"
+    if scale == exp.UnixToTime.MICROS:
+        return f"TIMESTAMP_MICROS({timestamp})"
+
+    raise ValueError("Improper scale for timestamp")
```

Comment:
You need to use `self.unsupported` so that the error level will be respected, e.g. if it's set to `IGNORE`

```suggestion
    self.unsupported("Improper scale for timestamp")
    return ""
```